meson.build

```diff
@@ -0,0 +1,39 @@
+project('notekit', 'cpp',
+	version : '0.1',
+	default_options : ['warning_level=3', 'cpp_std=c++14']
+)
+
+conf_data = configuration_data()
+conf_data.set('HAVE_CLATEXMATH', true)
+
+deps = [
+	dependency('gtkmm-3.0'),
+	dependency('gtksourceviewmm-3.0'),
+	dependency('jsoncpp'),
+	dependency('zlib'),
+	dependency('fontconfig'),
+	dependency('clatexmath')
```

Comment:
Could you make this an optional dependency? And have HAVE_CLATEXMATH being defined as true or false according to whether or not CLaTeXMath is present or not?

In Fedora, for example, it is impossible to add CLaTeXMath for licensing issues, so that would make it universal, as well as allowing this to be merged before the commit you pointed to was merged.

According to [Mesons' docs](https://mesonbuild.com/Dependencies.html), this could be done this way:
```
clatexmath = dependency('clatexmath', required : false)
```
and
```
if clatexmath.found()
   conf_data.set('HAVE_CLATEXMATH', true) 
else
   conf_data.set('HAVE_CLATEXMATH', false) 
endif

deps += clatexmath
```

Or something like this. I know nothing about meson (except that it had caused me way less problem than cmake). I don't even know if `deps += clatexmath` does work :stuck_out_tongue:...

Reply:
I had it like this originally, but I couldn't be bothered to write a python script that adds/removes `<context ref="latex"/>` in `sourceview/markdown.lang.in` because configure_file only works for c/c++ sourcecode in the main directory.

Because of that I thought, that it might not be a bad idea to have clatexmath as a "normal" dependency and have it included with notekit by default. 

> In Fedora, for example, it is impossible to add CLaTeXMath for licensing issues, so that would make it universal, as well as allowing this to be merged before the commit you pointed to was merged.

May I ask what license is the offending one, because I'd consider every license I've found (MIT / OFL/ Knuth / dsrom?) free "software".

Reply:
> May I ask what license is the offending one, because I'd consider every license I've found (MIT / OFL/ Knuth / dsrom?) free "software".
I don't remember the details well, but, IIRC, some license font wasn't documented by the Fedora Project. Basically, Fedora has extremely strict licensing guidelines. If some license wasn't vetted prior, its use is strictly prohibited. This includes COPR.

Thinking about it, I may try in the future to provide an external repo that enables CLaTeXMath for Fedora.

The only way to solve this would be to ask legal, but I can only do that if the Fedora package is submitted to official repos, COPR not being one, that is currently impossible :/...

I'd have to investigate this further.

Reply:
I just checked, it was dsrom.

Here is the list of all allowed licenses: https://fedoraproject.org/wiki/Licensing:Main

Reply:
It counts as a ["good license"](https://fedoraproject.org/wiki/Licensing:Main#Good_Licenses_4) why is that not allowed?

I mean, i agree that from a legal standpoint is is almost as bad as the WTFPL because it is not well written, but It should count as a free license.

Reply:
It's not that it's not allowed, it just hasn't been vetted yet ;) ! I just need to specifically request it. And since notekit is only in COPR for now, I can't do anything yet :/...

Reply:
> It's not that it's not allowed, it just hasn't been vetted yet ;) ! I just need to specifically request it.

I don't know much about the "Fedora Packaging Ecosystem", but I don't quite get what the problem is here.

- Make a clatexmath package that provides libclatexmath0_0-0 and clatexmath-devel (or whatever package naming spec fedora uses)
- Let the notekit package depend on it.

if the license is allowed, why would you have to request it?

Reply:
> if the license is allowed, why would you have to request it?

I don't know why, and I quite agree with that. But I guess that the project fears that just giving that to the best guess of the packager might be a bad idea, and they want lawyers involved to make sure that they are not messing up anything, even if it is obvious that it is free?

But quoting that same page:
> All software in Fedora must be under licenses in the Fedora licensing list _(the web page in sent)_.

Reply:
But it is in that list?

Reply:
Oh! It is the DoubleStroke License :laughing:! Thank you for pointing this out :stuck_out_tongue:! I was searching for `dsrom`, literally.

Reply:
Yeah, I see why the file is a bit confusing. @NanoMichael has probably just named the file like this.

Tip: if you have a license where you don't know the name just search one line of the contents.
In this case "You may use and distribute these fonts as you like." you even get the [fedora page](https://fedoraproject.org/wiki/Licensing:Fonts/DoubleStroke?rd=Licensing/Fonts/DoubleStroke) as the first result.